core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -187,7 +187,7 @@ class TransactionCoordinator(brokerId: Int,
     } else {
       // caller should have synchronized on txnMetadata already
       txnMetadata.state match {
-        case PrepareAbort | PrepareCommit =>
+        case PrepareAbort | PrepareCommit | Dead =>
```

Comment:
I think we should never see `Dead` as it is only set in the pending state but never in the state field.